salt/_modules/metalk8s.py

```diff
@@ -32,6 +32,25 @@ def deps_missing(*args, **kwargs):
     raise CommandExecutionError("Kubernetes python client is not installed")
 
 
+@depends('KUBERNETES_PRESENT', fallback_function=deps_missing)
+def wait_apiserver(retry=10, **kwargs):
+    """Wait for kube-apiserver to respond.
+
+    Simple "retry" wrapper around the kubernetes.ping Salt execution function.
+    """
+    status = False
+    attempts = 0
+    while not status and attempts < retry:
```

Comment:
This busy-loops, no? Shouldn't there be some kind of wait/sleep somewhere?

Reply:
I didn't check, but from what I remember in logs, `kubernetes.ping` does retry and wait a little between retries. But it makes senses to wait here as well, I'll add something.